### PR TITLE
Task 238

### PR DESCRIPTION
**Pull request by john-kheir**
fixes https://github.com/0-complexity/G8_testing/issues/238

### CODE REVIEW

File: functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py [hunk 1]
```diff
@@ -73,20 +74,52 @@ def test003_check_connectivity_through_external_network(self):
...
         #. Write a big file FS1 on VM1
         #. Migrate VM1 in the middle of writing a file, should succeed
         #. Check if the file has been written correctly after vm live migration
         """
-        # Note: this testcase may be hard to be implemented from here.
+        self.lg('%s STARTED' % self._testID)
+
+        self.lg("Create VM1,should succeed.")
+        vm1_id = self.cloudapi_create_machine(cloudspace_id=self.cloudspace_id)
+        self.assertTrue(vm1_id)
```
Comment (ahmedelsayed-93) [inline functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py]: no need for this line, `cloudapi_create_machine` method already assert for the machine id
https://github.com/0-complexity/G8_testing/blob/dd1c6dc70c64eefdae6ce9916c5d0535efd3ae23/functional_testing/Openvcloud/utils/utils.py#L225

File: functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py [hunk 2]
```diff
@@ -73,20 +74,52 @@ def test003_check_connectivity_through_external_network(self):
...
+        t.start()
+        time.sleep(7)
+
+        self.lg("Migrate VM1 to another node, should succeed.")
+        self.api.cloudbroker.machine.moveToDifferentComputeNode(machineId=vm1_id, reason="test",
+                                                                targetStackId=second_node, force=False)
+        self.assertEqual(self.api.cloudbroker.machine.get(machineId=vm1_id)["stackId"], second_node)
+
+        self.lg("Make sure that VM1 is running.")
+        self.assertEqual(self.api.cloudapi.machines.get(machineId=vm1_id)['status'], 'RUNNING')
```
Comment (ahmedelsayed-93) [inline functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py]: you can call this method only one time and assert for the filed you need

Comment (ahmedelsayed-93) [reply functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py]: just call this func `self.api.cloudbroker.machine.get` only on time and assert for the status and the stackid

File: functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py [hunk 3]
```diff
@@ -73,20 +74,52 @@ def test003_check_connectivity_through_external_network(self):
...
+        self.lg('Write a big file FS1 on VM1')
+        current_stackId = self.api.cloudbroker.machine.get(machineId=vm1_id)["stackId"]
+        second_node = self.get_running_stackId(current_stackId)
+        if not second_node:
+            self.skipTest('[*] No running nodes ')
+        vm1_conn = self.get_vm_connection(vm1_id)
+        cmd = "yes 'Some text' | head -n 200000000 > largefile.txt"
+        t = threading.Thread(target=vm1_conn.run, args=(cmd, ))
+        t.start()
+        time.sleep(7)
```
Comment (ahmedelsayed-93) [inline functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py]: why sleep ?

File: functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py (hunk 2 above)
Comment (john-kheir) [reply functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py]: done

File: functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py (hunk 1 above)
Comment (john-kheir) [reply functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py]: done

File: functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py (hunk 3 above)
Comment (john-kheir) [reply functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py]: To make sure the file is started to be written and some data are already there.

**Pull request merged**